lib/ret.ex

```diff
@@ -16,4 +21,76 @@ defmodule Ret do
       {:error, :forbidden}
     end
   end
+
+  defp delete_avatars_for_account(%Account{} = account) do
+    account_avatars_query = from(avatar in Avatar, where: avatar.account_id == ^account.account_id)
+    account_avatars = Repo.all(account_avatars_query) |> Repo.preload(Avatar.file_columns())
+    account_avatar_owned_files = Enum.flat_map(account_avatars, &avatar_owned_files/1)
+    Repo.delete_all(account_avatars_query)
+    delete_owned_files(account_avatar_owned_files)
+  end
+
+  defp avatar_owned_files(%Avatar{} = avatar) do
+    for column <- Avatar.file_columns(),
+        owned_file = Map.fetch!(avatar, column),
```

Comment:
@brianpeiris Does this fetch return a list of owned files?

Reply:
Nope, each column represents a single owned file, but each avatar has multiple columns that reference owned files, so this function will return a flat array of files for a single avatar.